packages/block-editor/src/components/gradient-picker/control.js

```diff
@@ -17,11 +17,14 @@ import { useSelect } from '@wordpress/data';
  */
 import GradientPicker from './';
 
-export default function( { className, label = __( 'Gradient Presets' ), ...props } ) {
-	const gradients = useSelect( ( select ) => (
-		select( 'core/block-editor' ).getSettings().gradients
+export default function( { className, value, onChange, label = __( 'Gradient Presets' ), ...props } ) {
+	const { gradients = [], disableCustomGradients } = useSelect( ( select ) => (
+		pick(
+			select( 'core/block-editor' ).getSettings(),
+			[ 'gradients', 'disableCustomGradients' ]
+		)
 	) );
-	if ( isEmpty( gradients ) ) {
+	if ( isEmpty( gradients ) && disableCustomGradients ) {
```

Comment:
Shouldn't this be an or not an and

Reply:
It is supposed to be an "and". Users can not choose a gradient if there are no presets and they can not use the custom gradient picker. If there are presets, or custom gradients are enabled the component should render something.

Reply:
I think that `disableCustomGradient` is set, then it should ignore defined gradients. ATM `disableCustomColors` has to set to true and the list of gradients has to be empty.

Reply:
Sorry, I'm not managing the clarify this logic.  If there are gradients available, the gradient picker should render even if custom gradients are disabled to allow the user to choose a preset gradient. If disableCustomGradients is false, the gradient picker should render even if the gradients list is empty to allow the user to create a custom gradient.

The only case where the gradient picker should not render is if the gradients list is empty and custom gradients are disabled because, in that case, the user cannot select a preset gradient or create a custom gradient.